vm/tests/runner.rs

```diff
@@ -0,0 +1,289 @@
+use std::path::PathBuf;
+
+use analyzer::importer::{ASTImporter, ImportResult, StaticImporter};
+use analyzer::name::Name;
+use analyzer::reef::{Externals, Reef};
+use analyzer::relations::SourceId;
+use analyzer::types::hir::ExprKind;
+use analyzer::types::ty::{Type, TypeRef};
+use analyzer::{analyze, types, Analyzer, Inject};
+use cli::pipeline::FileImporter;
+use compiler::{compile, CompilerOptions};
+use context::source::Source;
+use parser::parse_trusted;
+use vm::value::VmValue;
+use vm::{VmError, VmValueFFI, VM};
+
+pub struct Runner<'a> {
+    externals: Externals<'a>,
+    vm: VM,
+    analyzer: Analyzer<'a>,
+}
+
+impl Default for Runner<'_> {
+    fn default() -> Self {
+        let mut externals = Externals::default();
+        let mut std_importer = FileImporter::new(PathBuf::from("../lib"));
+        let mut vm = VM::default();
+
+        let std_name = Name::new("std");
+        let analyzer = analyze(std_name.clone(), &mut std_importer, &externals);
+        let mut buff = Vec::new();
+
+        compile(
+            &analyzer.engine,
+            &analyzer.typing,
+            &analyzer.resolution.relations,
+            &analyzer.resolution.engine,
+            &externals,
+            externals.current,
+            SourceId(0),
+            &mut buff,
+            CompilerOptions::default(),
+        )
+        .expect("std did not compile successfully");
+
+        vm.register(&buff).expect("VM std register");
+        unsafe {
+            vm.run().expect("VM std init");
+        }
+
+        externals.register(Reef::new("std".to_string(), analyzer));
+
+        Self {
+            externals,
+            vm,
+            analyzer: Analyzer::default(),
+        }
+    }
+}
+
+impl<'a> Runner<'a> {
+    pub fn eval(&mut self, expr: &'a str) -> VmValue {
+        match self.try_eval(expr) {
+            Ok(v) => v,
+            Err(VmError::Panic) => panic!("VM did panic"),
+            Err(VmError::Internal) => panic!("VM internal error"),
+        }
+    }
+
+    pub fn try_eval(&mut self, expr: &'a str) -> Result<VmValue, VmError> {
```

Comment:
As those tests are closely related to the VM, you may want to verify that the stack is empty after the execution.